### PR TITLE
Fix edmac

### PR DESCRIPTION
**Pull request by hzhuang1**

### CODE REVIEW

**Comment by hzhuang1** [comment]
Fix to init EDMAC & RTC in HiKey platform.

**Comment by hzhuang1** [comment]
Yes, I verified it.

**Comment by davidcunado-arm** [comment]
@hzhuang1 
Excuse my ignorance... what is the origin of the EDMA controller IP point? Is it a publicly available IP? Is there a link to the TRM?

**Comment by hzhuang1** [comment]
EDMA is the generic DMA controller of low speed peripherals that belongs to Hisilicon. The document is in https://github.com/96boards/documentation/blob/master/ConsumerEdition/HiKey/HardwareDocs/Hi6220V100_Multi-Mode_Application_Processor_Function_Description.pdf

But we don't have the full document on EDMA. This document only describes registers and channels. And it doesn't describe how it work with secure world. I think this initialization only decides routing DMA channels to secure world or non-secure world.

**Comment by hzhuang1** [comment]
@davidcunado-arm Could you help to merge this fix? We're pending on it.

**Comment by davidcunado-arm** [comment]
@hzhuang1 
Apologies, we have a bit of a backlog of items - I'll get to this today or Monday at the latest. Sorry for the delay :(

File: plat/hisilicon/hikey/hikey_bl31_setup.c [hunk 1]
```diff
@@ -152,6 +153,20 @@ void bl31_plat_arch_setup(void)
...
+static void hikey_edma_init(void)
+{
+	int i;
+	uint32_t non_secure;
+
+	non_secure = EDMAC_SEC_CTRL_INTR_SEC | EDMAC_SEC_CTRL_GLOBAL_SEC;
+	mmio_write_32(EDMAC_SEC_CTRL, non_secure);
+
+	for (i = 0; i < EDMAC_CHANNEL_NUMS; i++) {
+		mmio_write_32(EDMAC_AXI_CONF(i), (1 << 6) | (1 << 18));
```
Comment (davidcunado-arm) [inline plat/hisilicon/hikey/hikey_bl31_setup.c]: For (1 << 6 ) and (1 <<18) - perhaps having these as defines in hi6220.h would be clearer.

**Comment by davidcunado-arm** [comment]
@hzhuang1 
My only concern here is that the EDMA changes are referring to changes that aren't public.

The EDMA IP belongs to Hisilison, right? If so, can you confirm that there are no IP issues?

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Pull request merged**